doc/rtd/reference/datasources/nocloud.rst

```diff
@@ -4,97 +4,237 @@ NoCloud
+user to configure an instance.
+
+meta-data
+---------
+
+The ``meta-data`` file is a YAML-formatted file. This is a required
+configuration type.
+
+vendor-data (optional)
+----------------------
+
+Vendor-data may be used to provide default cloud-specific configurations which
+may be overriden by user-data. This may be useful, for example, configure an
+instance with a cloud provider's repository mirror for faster package
+installation.
+
+network-config (optional)
+-------------------------
+
+Network configuration typically comes from the cloud provider to set
+cloud-specific network configurations or a reasonable default is set by
+cloud-init (typically cloud-init brings up an interface using DHCP).
+
+Since NoCloud is a generic datasource, network configuration may be set the
+same way as user-data, meta-data, vendor-data.
 
-Method 1: Labeled filesystem
+See the :ref:`network configuration<network_config>` documentation for
+information on network configuration formats.
+
+Meta-configuration
+==================
+
+The purpose of the meta-configuration is to tell cloud-init where it can find
+the runtime configurations described above.
+
+There are two methods for cloud-init to receive a meta-configuration.
+
+Method 1: Line configuration
 ----------------------------
 
-A labeled `vfat`_ or `iso9660` filesystem may be used. The filesystem volume
-must be labelled ``CIDATA``.
+The "line configuration" is a single string of text which is passed to an
+instance at boot time via either the kernel command line or in the serial
+number exposed via DMI (sometimes called SMBIOS).
 
+Example: ::
 
-Method 2: Custom webserver
---------------------------
+  ds=nocloud;s=https://10.42.42.42/configs/
 
-Configuration files can be provided to cloud-init over HTTP(s). To tell
-cloud-init the URI to use, arguments must be passed to the instance via the
-kernel command line or SMBIOS serial number. This argument might look like: ::
+In the above line configuration, ``ds=nocloud`` tells cloud-init to use the
+NoCloud datasource, and ``s=https://10.42.42.42/configs/`` tells cloud-init to
+fetch configurations using ``https`` from the URI
+``https://10.42.42.42/configs/``.
 
-  ds=nocloud;s=https://10.42.42.42/cloud-init/configs/
+We will describe the possible values in a line configuration in the following
+sections.
 
 .. note::
+
    If using kernel command line arguments with GRUB, note that an
    unescaped semicolon is intepreted as the end of a statement.
-   Consider using single-quotes to avoid this pitfall. See: `GRUB quoting`_
-   ds=nocloud;s=http://10.42.42.42/cloud-init/configs/
+   See: `GRUB quoting`_
+
+Method 2: System configuration
+------------------------------
+
+System configurations are YAML-formatted files and have names that end in
+``.cfg``. These are located under :file:`/etc/cloud/cloud.cfg.d/`.
+
+Example:
+
+.. code-block:: yaml
+
+  datasource:
+    NoCloud:
+      seedfrom: https://10.42.42.42/configs/
```

Comment:
I think this section needs an additional space indentation to make the code block render correctly